### PR TITLE
feat(base-cluster/tetragon): add tetragon

### PR DESCRIPTION
**Pull request by marvinWolff**
Ref.: https://github.com/teutonet/teutonet-helm-charts/issues/1046

### CODE REVIEW

File: charts/base-cluster/values.yaml [hunk 1]
```diff
@@ -28,6 +28,10 @@ global:
       condition: "{{ .Values.ingress.enabled }}"
       additionalLabels:
         app.kubernetes.io/component: ingress
+    tetragon:
+      condition: "{{ .Values.tetragon.enabled }}"
+      additionalLabels:
+        app.kubernetes.io/component: tetragon
```
Comment (cwrau) [inline charts/base-cluster/values.yaml:34]: I would rather put this in `kube-system`

File: charts/base-cluster/values.yaml [hunk 2]
```diff
@@ -457,3 +465,9 @@ backup:
 
 kube-janitor:
   enabled: false
+
+tetragon:
+  enabled: true
+  serviceMonitor:
+    labelsOverride:
+      managed.by/monitoring: teutonet
```
Comment (cwrau) [inline charts/base-cluster/values.yaml:473]: Definitely not, we're not doing this anywhere else anyways

```suggestion
```

File: charts/base-cluster/values.schema.json [hunk 3]
```diff
@@ -549,6 +549,30 @@
...
+              "type": "object",
+              "properties": {
+                "labelsOverride": {
+                  "additionalProperties": {
+                    "type": "string"
+                  }
+                }
+              }
+            }
+          }
```
Comment (cwrau) [inline charts/base-cluster/values.schema.json:571]:
```suggestion
```

File: charts/base-cluster/templates/tetragon/tetragon.yaml [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    tetragonOperator:
+      enabled: false
+    tetragon:
+      enableProcessCred: true
+      enableProcessNs: true
+      {{- if .Values.monitoring.prometheus.enabled }}
+      prometheus:
+        serviceMonitor:
+          enabled: true
+          labelsOverride: {{- toYaml .Values.tetragon.serviceMonitor.labelsOverride | nindent 12 }}
```
Comment (cwrau) [inline charts/base-cluster/templates/tetragon/tetragon.yaml:35]:
```suggestion
          labelsOverride: {{- toYaml .Values.monitoring.labels | nindent 12 }}
```

**Review by cwrau — CHANGES_REQUESTED** [review]

File: charts/base-cluster/values.yaml (hunk 1 above)
Comment (marvinWolff) [reply charts/base-cluster/values.yaml]: done

File: charts/base-cluster/values.yaml [hunk 5]
```diff
@@ -141,6 +145,10 @@ global:
         kyverno: 3.x.x
         kyverno-policies: 3.x.x
       condition: "{{ .Values.kyverno.enabled }}"
+    tetragon:
+      url: https://helm.cilium.io
+      charts:
+        tetragon: 1.x.x
```
Comment (cwrau) [inline charts/base-cluster/values.yaml:151]:
```suggestion
        tetragon: 1.x.x
      condition: "{{ .Values.tetragon.enabled }}"
```

File: charts/base-cluster/templates/monitoring/kube-prometheus-stack/_prometheus-stack-config.yaml [hunk 6]
```diff
@@ -1,7 +1,7 @@
 {{- define "base-cluster.prometheus.config" -}}
   {{- if .Values.global.imageRegistry }}
 global:
-  imageRegistry: {{ .Values.global.imageRegistry }}
+    imageRegistry: {{ .Values.global.imageRegistry }}
```
Comment (cwrau) [inline charts/base-cluster/templates/monitoring/kube-prometheus-stack/_prometheus-stack-config.yaml:4]: Why?

```suggestion
  imageRegistry: {{ .Values.global.imageRegistry }}
```

File: charts/base-cluster/templates/tetragon/tetragon.yaml [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+  {{- if .Values.monitoring.prometheus.enabled }}
+  dependsOn:
+    - name: kube-prometheus-stack
+      namespace: monitoring
+  {{- end }}
+  values:
+    export:
+      stdout:
+        image:
+          repository: {{ .Values.global.imageRegistry }}
```
Comment (cwrau) [inline charts/base-cluster/templates/tetragon/tetragon.yaml:24]: Why did you change this from `repository: {{ printf "%s/cilium/hubble-export-stdout" ($.Values.global.imageRegistry | default (include "base-cluster.defaultRegistry" (dict))) }}`?

File: charts/base-cluster/templates/tetragon/tetragon.yaml [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+        capabilities:
+          drop:
+            - ALL
+    crds:
+      installMethod: helm
+    tetragonOperator:
+      enabled: false
+    tetragon:
+      image:
+        repository: {{ .Values.global.imageRegistry }}
```
Comment (cwrau) [inline charts/base-cluster/templates/tetragon/tetragon.yaml:35]: Why did you change this from `repository: {{ printf "%s/cilium/tetragon" ($.Values.global.imageRegistry | default (include "base-cluster.defaultRegistry" (dict))) }}`?

File: charts/base-cluster/templates/monitoring/kube-prometheus-stack/_prometheus-stack-config.yaml (hunk 6 above)
Comment (marvinWolff) [reply charts/base-cluster/templates/monitoring/kube-prometheus-stack/_prometheus-stack-config.yaml]: Was a typo

File: charts/base-cluster/templates/tetragon/tetragon.yaml (hunk 7 above)
Comment (marvinWolff) [reply charts/base-cluster/templates/tetragon/tetragon.yaml]: changed it to an `if` statement

File: charts/base-cluster/templates/tetragon/tetragon.yaml (hunk 8 above)
Comment (marvinWolff) [reply charts/base-cluster/templates/tetragon/tetragon.yaml]: changed it to an if statement

**Review by cwrau — APPROVED** [review]
